pkg/scheduler/predicates/ha.go

```diff
@@ -127,13 +142,69 @@ func (h *ha) Filter(instanceName string, pod *apiv1.Pod, nodes []apiv1.Node) ([]
 	return getNodeFromNames(nodes, minNodeNames), nil
 }
 
+func (h *ha) realAcquireLock(pod *apiv1.Pod) (*apiv1.PersistentVolumeClaim, *apiv1.PersistentVolumeClaim, error) {
+	ns := pod.GetNamespace()
+	component := pod.Labels[label.ComponentLabelKey]
+	instanceName := pod.Labels[label.InstanceLabelKey]
+	podName := pod.GetName()
+	pvcList, err := h.pvcListFn(ns, instanceName, component)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	currentPVCName := pvcName(component, podName)
+	var currentPVC *apiv1.PersistentVolumeClaim
+	var schedulingPVC *apiv1.PersistentVolumeClaim
+	for i := 0; i < len(pvcList.Items); i++ {
+		pvc := pvcList.Items[i]
+		if pvc.GetName() == currentPVCName && currentPVC == nil {
```

Comment:
Why do you want to add the condition `currentPVC == nil`, will there be multiple PVCs with the same name?

Reply:
Removed.